plugins/net_plugin/net_plugin.cpp

```diff
@@ -880,7 +881,13 @@ namespace eosio {
          uint32_t end   = std::max( peer_requested->end_block, block_header::num_from_id(head_id) );
          peer_requested = sync_state( start, end, start - 1 );
       }
-      enqueue_sync_block();
+
+      if( peer_requested->start_block <= peer_requested->end_block ) {
```

Comment:
Same for this

Reply:
https://github.com/EOSIO/eos/pull/7625/files#diff-3935dcf0fa19e1c59296f2ff874e2621R1036